<?xml version="1.0" encoding="utf-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
  <file datatype="xml" original="ErrorAddDataBoundRowEventArgs.xml" source-language="en-US" target-language="zh-CN">
    <header>
      <tool tool-id="mdxliff" tool-name="mdxliff" tool-version="1.0-38f25dd" tool-company="Microsoft" />
      <xliffext:skl_file_name xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">b4690f9a-9e49-426d-9091-855b10fda4db018b6377ebcf058e3a74c0091371a756b82e61b1.skl</xliffext:skl_file_name>
      <xliffext:version xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">1.2</xliffext:version>
      <xliffext:ms.openlocfilehash xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">018b6377ebcf058e3a74c0091371a756b82e61b1</xliffext:ms.openlocfilehash>
      <xliffext:ms.sourcegitcommit xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">bb8a3a96823cedeffbdbcc0c41a8118399de8074</xliffext:ms.sourcegitcommit>
      <xliffext:ms.lasthandoff xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">02/21/2018</xliffext:ms.lasthandoff>
      <xliffext:moniker_ids xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">vsto-2017</xliffext:moniker_ids>
    </header>
    <body>
      <group id="content" extype="content">
        <trans-unit id="101" translate="yes" xml:space="preserve" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>Provides information for the <ph id="ph1">&lt;see cref="E:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow" /&gt;</ph> event.</source>
          <target state="translated">提供有关 <ph id="ph1">&lt;see cref="E:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow" /&gt;</ph> 事件的信息。</target>       </trans-unit>
        <trans-unit id="102" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>The <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph> event is raised by the <ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph> control if an exception occurs during an attempt to add a new row to the data source bound to the <ph id="ph3">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>.</source>
          <target state="translated"><ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph>事件由引发<ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>如果尝试将新行添加到数据源过程中发生异常的控件绑定到<ph id="ph3">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>。</target>       </trans-unit>
        <trans-unit id="103" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>The <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item%2A&gt;</ph> property can be used to get the new item that the <ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph> tried to add.</source>
          <target state="translated"><ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item%2A&gt;</ph>属性可以用于获取新项<ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>尝试添加。</target>       </trans-unit>
        <trans-unit id="104" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>The <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A&gt;</ph> property provides the exception that was thrown, and the <ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry%2A&gt;</ph> property can be used to indicate whether another attempt to add this row should be made.</source>
          <target state="translated"><ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A&gt;</ph>属性提供已引发的异常和<ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry%2A&gt;</ph>属性可以用于指示是否应进行另一个尝试添加此行。</target>       </trans-unit>
        <trans-unit id="105" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>The following code example creates a <ph id="ph1">&lt;xref:System.Data.DataTable&gt;</ph> and a <ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>, and binds the <ph id="ph3">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph> to the <ph id="ph4">&lt;xref:System.Data.DataTable&gt;</ph>.</source>
          <target state="translated">下面的代码示例创建<ph id="ph1">&lt;xref:System.Data.DataTable&gt;</ph>和<ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>，并将绑定<ph id="ph3">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>到<ph id="ph4">&lt;xref:System.Data.DataTable&gt;</ph>。</target>       </trans-unit>
        <trans-unit id="106" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>It then creates an <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph> event handler.</source>
          <target state="translated">然后，它创建<ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph>事件处理程序。</target>       </trans-unit>
        <trans-unit id="107" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>To test the event, manually add a new row to the <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph> and enter the last name "Chan" and a first name.</source>
          <target state="translated">若要测试事件，手动向其中添加新行<ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>输入姓氏"通道"和第一个名称。</target>       </trans-unit>
        <trans-unit id="108" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>The event handler displays a message.</source>
          <target state="translated">事件处理程序会显示一条消息。</target>       </trans-unit>
        <trans-unit id="109" translate="yes" xml:space="preserve" extradata="MT" uid="T:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs">
          <source>This example is for a document-level customization.</source>
          <target state="translated">此示例适用于文档级自定义。</target>       </trans-unit>
        <trans-unit id="110" translate="yes" xml:space="preserve" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException">
          <source>Gets the <ph id="ph1">&lt;see cref="T:System.Exception" /&gt;</ph> instance that caused the current exception.</source>
          <target state="translated">获取导致当前异常的 <ph id="ph1">&lt;see cref="T:System.Exception" /&gt;</ph> 实例。</target>       </trans-unit>
        <trans-unit id="111" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException">
          <source>An instance of <ph id="ph1">&lt;see cref="T:System.Exception" /&gt;</ph> that describes the error that caused the current exception.</source>
          <target state="translated">一个 <ph id="ph1">&lt;see cref="T:System.Exception" /&gt;</ph> 的实例，描述导致当前异常的错误。</target>       </trans-unit>
        <trans-unit id="112" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException">
          <source>The <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph> event is raised as a direct result of an exception while attempting to add a new row to the data source.</source>
          <target state="translated"><ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph>作为尝试将新行添加到数据源时出现异常的直接结果引发事件。</target>       </trans-unit>
        <trans-unit id="113" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException">
          <source>Use the <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A&gt;</ph> property to obtain the exception that led to this error.</source>
          <target state="translated">使用<ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException%2A&gt;</ph>属性来获取与此错误引发的异常。</target>       </trans-unit>
        <trans-unit id="114" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.InnerException">
          <source>For example, if the <ph id="ph1">&lt;xref:System.Data.DataTable&gt;</ph> that is bound to the <ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph> control does not allow a column to be <ph id="ph3">`null`</ph>, and the user does not specify a value for this column, the attempt to add the row causes a <ph id="ph4">&lt;xref:System.Data.NoNullAllowedException&gt;</ph> to be raised.</source>
          <target state="translated">例如，如果<ph id="ph1">&lt;xref:System.Data.DataTable&gt;</ph>，它绑定到<ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>控件不允许将一个列<ph id="ph3">`null`</ph>，和用户未指定此列的值，尝试将该行添加导致<ph id="ph4">&lt;xref:System.Data.NoNullAllowedException&gt;</ph>被引发。</target>       </trans-unit>
        <trans-unit id="115" translate="yes" xml:space="preserve" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item">
          <source>Gets the new item that the <ph id="ph1">&lt;see cref="T:Microsoft.Office.Tools.Excel.ListObject" /&gt;</ph> attempted to add to the data source.</source>
          <target state="translated">获取 <ph id="ph1">&lt;see cref="T:Microsoft.Office.Tools.Excel.ListObject" /&gt;</ph> 尝试添加到数据源的新项。</target>       </trans-unit>
        <trans-unit id="116" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item">
          <source>The item to be added.</source>
          <target state="translated">要添加的项。</target>       </trans-unit>
        <trans-unit id="117" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Item">
          <source>When the <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph> is bound to a <ph id="ph2">&lt;xref:System.Data.DataTable&gt;</ph>, the item would be a <ph id="ph3">&lt;xref:System.Data.DataRowView&gt;</ph> rather than a <ph id="ph4">&lt;xref:System.Data.DataRow&gt;</ph>.</source>
          <target state="translated">当<ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>绑定到<ph id="ph2">&lt;xref:System.Data.DataTable&gt;</ph>，该项应当是<ph id="ph3">&lt;xref:System.Data.DataRowView&gt;</ph>而不是<ph id="ph4">&lt;xref:System.Data.DataRow&gt;</ph>。</target>       </trans-unit>
        <trans-unit id="118" translate="yes" xml:space="preserve" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry">
          <source>Gets or sets a value indicating whether the <ph id="ph1">&lt;see cref="T:Microsoft.Office.Tools.Excel.ListObject" /&gt;</ph> should attempt to add the item again.</source>
          <target state="translated">获取或设置一个值，该值指示 <ph id="ph1">&lt;see cref="T:Microsoft.Office.Tools.Excel.ListObject" /&gt;</ph> 是否应该再次尝试添加此项。</target>       </trans-unit>
        <trans-unit id="119" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry">
          <source><ph id="ph1">&lt;see langword="true" /&gt;</ph> if another attempt should be made; otherwise, <ph id="ph2">&lt;see langword="false" /&gt;</ph>.</source>
          <target state="translated"><ph id="ph1">&lt;see langword="true" /&gt;</ph> 如果应进行另一个尝试;否则为<ph id="ph2">&lt;see langword="false" /&gt;</ph>。</target>       </trans-unit>
        <trans-unit id="120" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry">
          <source>The default value is <ph id="ph1">&lt;see langword="false" /&gt;</ph>.</source>
          <target state="translated">默认值为 <ph id="ph1">&lt;see langword="false" /&gt;</ph>。</target>       </trans-unit>
        <trans-unit id="121" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry">
          <source>If the <ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph> encounters an error when attempting to add the new row, the <ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph> event is raised.</source>
          <target state="translated">如果<ph id="ph1">&lt;xref:Microsoft.Office.Tools.Excel.ListObject&gt;</ph>添加新行中，在尝试时遇到错误<ph id="ph2">&lt;xref:Microsoft.Office.Tools.Excel.ListObject.ErrorAddDataBoundRow&gt;</ph>引发事件。</target>       </trans-unit>
        <trans-unit id="122" translate="yes" xml:space="preserve" extradata="MT" uid="P:Microsoft.Office.Tools.Excel.ErrorAddDataBoundRowEventArgs.Retry">
          <source>The developer can handle this event and attempt to fix any errors that occur.</source>
          <target state="translated">开发人员可以处理此事件，并尝试解决出现的任何错误。</target>       </trans-unit>
      </group>
    </body>
  </file>
</xliff>